synapse/config/key.py

```diff
@@ -59,21 +59,31 @@ def read_config(self, config):
 
         self.expire_access_token = config.get("expire_access_token", False)
 
+        # a secret which is used to calculate HMACs for form values, to stop
+        # falsification of values
+        self.form_secret = config.get("form_secret", None)
+
     def default_config(self, config_dir_path, server_name, is_generating_file=False,
                        **kwargs):
         base_key_name = os.path.join(config_dir_path, server_name)
 
         if is_generating_file:
             macaroon_secret_key = random_string_with_symbols(50)
+            form_secret = '"%s"' % random_string_with_symbols(50)
         else:
             macaroon_secret_key = None
+            form_secret = 'null'
```

Comment:
Why are `macroon_secret_key` and `form_secret` handled differently?

Reply:
Actually I think the `macaroon_secret_key` code is buggy. You end up with an explicit "None" as the secret key :/

Reply:
If its easy to fix up then that would be good, as otherwise I reckon its going to cause confusion in the future.